frontend/schemaCMS/src/modules/dataSource/dataSource.sagas.js

```diff
@@ -293,6 +293,38 @@ function* updateOne({ payload: { requestData, dataSource } }) {
       return;
     }
 
+    if (
+      requestData[DATA_SOURCE_API_URL] ||
+      requestData[DATA_SOURCE_API_JSON_PATH] ||
+      requestData[DATA_SOURCE_API_JSON_PATH] === '' ||
+      requestData[DATA_SOURCE_AUTO_REFRESH] ||
+      requestData[DATA_SOURCE_AUTO_REFRESH] === false
+    ) {
+      const { data } = yield api.patch(`${DATA_SOURCES_PATH}/${dataSource.id}`, {
+        [DATA_SOURCE_API_URL]: requestData[DATA_SOURCE_API_URL] || dataSource[DATA_SOURCE_API_URL],
+        [DATA_SOURCE_API_JSON_PATH]: requestData[DATA_SOURCE_API_JSON_PATH],
+        [DATA_SOURCE_AUTO_REFRESH]: requestData[DATA_SOURCE_AUTO_REFRESH],
+        [DATA_SOURCE_RUN_LAST_JOB]: requestData[DATA_SOURCE_RUN_LAST_JOB],
+        [DATA_SOURCE_TYPE]: requestData[DATA_SOURCE_TYPE],
+      });
+
+      browserHistory.push(`/project/${dataSource.project.id}/datasource`);
+      yield put(DataSourceRoutines.updateOne.success({ dataSource: data }));
+      return;
+    }
+
+    if (requestData[DATA_SOURCE_REIMPORT]) {
```

Comment:
If `DATA_SOURCE_API_URL` will be defined then probably we never enter this if because we land in the previous one.

Reply:
No, because requestData[DATA_SOURCE_REIMPORT] is not defined when you click different button, so these are 2 separate conditions. Also in previous condition requestData[DATA_SOURCE_API_URL] is taken under consideration, here we're looking at dataSource[DATA_SOURCE_API_URL].
To simplify: 
- previous condition is about `if (form fields changed and save button clicked)`
- this condition is about `if (reimport button clicked)`